fbgemm_gpu/CMakeLists.txt

```diff
@@ -9,13 +9,27 @@ message("${message_line}")
 if(SKBUILD)
   message("The project is built using scikit-build")
 endif()
+
+if(EXISTS "/usr/bin/nvidia-smi")
+  message("NVIDIA GPU detected.")
+  option(USE_CUDA "Use CUDA" ON)
+  option(USE_ROCM "Use ROCm" OFF)
+elseif(EXISTS "/opt/rocm/bin/rocm-smi")
+  message("AMD GPU detected.")
+  option(USE_CUDA "Use CUDA" OFF)
+  option(USE_ROCM "Use ROCm" ON)
+else()
+  message("Unable to detect GPU vendor")
```

Comment:
Please check the test failure on the PR: `build_gpu` and `build_cpu_only` tests. We might install fbgemm on pure CPU machines (with --cpu-only option), or we might install fbgemm GPU on a pure CPU machine to test the build correctness (installing CUDA library on the pure CPU machine but not runnable).

Reply:
I've changed the logic of detecting the GPU vendor, and waiting for the test.

Reply:
The CI passed.